docs/docs/development/onboarding.md

```diff
@@ -0,0 +1,257 @@
+3. Guides
+    1. [TinyGo to Wasm with JS apis - compile a Wasm module and execute it inside a JS environment (browser)](https://tinygo.org/docs/guides/webassembly/wasm/)
+    2. [TinyGo to WASI](https://wasmbyexample.dev/home.en-us.html) - compile a Wasm blob and execute it inside another Go host application ([Wazero VM](https://wazero.io/))
+        1. https://github.com/tetratelabs/wazero/tree/main/examples/allocation - check README and `tinygo` folder
+        2. https://github.com/tetratelabs/wazero/tree/main/examples/import-go - check README and `tinygo` folder
+
+## 1.3. TinyGo fork 🧪
+
+We have [forked TinyGo](https://github.com/LimeChain/tinygo/) as we need to add a new target for the Polkadot-specific wasm blob, targeting standalone **Wasm MVP**, similar to Rust's `wasm32-unknown-unknown`, **without bulk memory operations and other extensions,** also incorporating **custom GC** that utilizes an external allocator. In the [polkawasm-target-dev branch](https://github.com/LimeChain/tinygo/tree/polkawasm-target-dev), you can see the changes for the specific TinyGo releases.
+
+- Example: [shows the changes](https://github.com/limechain/tinygo/compare/dev...polkawasm-target-dev) added to TinyGo `v0.30`.
+
+We use a local build of TinyGo and do not depend on the already-built brew dependency.
+
+[Here are the steps](https://github.com/LimeChain/gosemble/blob/develop/docs/docs/development/toolchain-setup.md) how to install and build it locally.
+
+After you have built TinyGo, execute the following:
+
+```bash
+tinygo version
+```
+
+The output should be similar to:
+
+```bash
+tinygo version 0.30.0-dev darwin/arm64 (using go version go1.21.6 and LLVM version 17.0.6)
+```
+
+# 2. Architecture of a Polkadot Node
+
+Now that you have learned about WebAssembly, shared memory, runtime imported/exported functions and the TinyGo toolchain, let’s look at the Polkadot specification.
+
+Polkadot node architecture and protocol specification is heavily influenced by the tech stack: WebAssembly MVP and Rust. Some implementation details, like the memory management, are not well abstracted and tightly coupled with the Rust implementation and even included as part of the protocol specification.
+
+## 2.1. Node 💻
+
+[Polkadot protocol](https://spec.polkadot.network/id-polkadot-protocol) has been divided into two parts, the [Polkadot Runtime](https://spec.polkadot.network/part-polkadot-runtime) and the [Polkadot Host](https://spec.polkadot.network/part-polkadot-host).
+
+## 2.2. Host-Runtime Interaction 🤝
+
+### Imported Functions 📥
+
+External functions provided by the **Host** environment (Substrate/Kagome/Gossamer host) that **Runtime** (WebAssembly module) can invoke when needed, for more details check the [Host API](https://spec.polkadot.network/chap-host-api). The Host API provides access to memory, storage, crypto, hashing, logging and misc functionality.
+
+Example (Storage):
+
+- Rust implementation using Substrate
```

Comment:
Maybe we've missed the link to the rust code here - https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/io/src/lib.rs#L172

Reply:
We can use commit hashes as branches lines might move up-down in the future